src/utils/rocksdb_serialization.hpp

```diff
@@ -106,65 +84,69 @@ inline std::vector<storage::LabelId> TransformFromStringLabels(std::vector<std::
-  return ser_labels;
-}
+inline std::string SerializeLabels(const std::vector<std::string> &labels) { return utils::Join(labels, ","); }
 
 inline std::string SerializeProperties(const storage::PropertyStore &properties) { return properties.StringBuffer(); }
 
-/// TODO: andi Probably it is better to add delimiter between label,property and the rest of labels
-/// TODO: reuse PutIndexingLabelAndPropertiesFirst
 inline std::string PutIndexingLabelAndPropertyFirst(const std::string &indexing_label,
                                                     const std::string &indexing_property,
                                                     const std::vector<std::string> &vertex_labels) {
-  std::string result = indexing_label + "," + indexing_property;
+  std::string result;
+  result += indexing_label;
+  result += ",";
+  result += indexing_property;
+
   for (const auto &label : vertex_labels) {
     if (label != indexing_label) {
-      result += "," + label;
+      result += ",";
+      result += label;
     }
   }
   return result;
 }
 
 inline std::string PutIndexingLabelAndPropertiesFirst(const std::string &target_label,
                                                       const std::vector<std::string> &target_properties) {
-  std::string result = target_label;
+  std::string result;
+  result += target_label;
   for (const auto &target_property : target_properties) {
-    result += "," + target_property;
+    result += ",";
+    result += target_property;
   }
   return result;
 }
 
 inline storage::Gid ExtractSrcVertexGidFromEdgeValue(const std::string value) {
-  const std::string_view src_vertex_gid_str = GetViewOfFirstPartOfSplit(value, '|');
-  return storage::Gid::FromString(src_vertex_gid_str);
+  return storage::Gid::FromString(FindPartOfStringView(value, '|', 1));
 }
 
 inline storage::Gid ExtractDstVertexGidFromEdgeValue(const std::string value) {
-  const std::string_view dst_vertex_gid_str = GetViewOfSecondPartOfSplit(value, '|');
-  return storage::Gid::FromString(dst_vertex_gid_str);
+  return storage::Gid::FromString(FindPartOfStringView(value, '|', 2));
 }
 
 inline storage::EdgeTypeId ExtractEdgeTypeIdFromEdgeValue(const std::string_view value) {
-  const std::string_view edge_type_str = GetViewOfThirdPartOfSplit(value, '|');
-  return storage::EdgeTypeId::FromString(edge_type_str);
+  return storage::EdgeTypeId::FromString(FindPartOfStringView(value, '|', 3));
+}
+
+inline std::string_view GetPropertiesFromEdgeValue(const std::string_view value) {
+  return FindPartOfStringView(value, '|', 4);
 }
 
 inline std::string SerializeEdgeAsValue(const std::string &src_vertex_gid, const std::string &dst_vertex_gid,
                                         const storage::EdgeTypeId &edge_type, const storage::Edge *edge = nullptr) {
-  auto tmp = src_vertex_gid + "|" + dst_vertex_gid + "|" + SerializeIdType(edge_type) + "|";
+  std::string edge_type_str = edge_type.ToString();
+  std::string result;
```

Comment:
Is this faster than the old way?

Reply:
I think yes because less intermediate strings